packages/next/src/build/index.ts

```diff
@@ -1238,6 +1238,16 @@ export default async function build(
 
       const { cacheHandler } = config
 
+      const instrumentationHookEntryFiles = hasInstrumentationHook
+        ? [
+            path.join(SERVER_DIRECTORY, `${INSTRUMENTATION_HOOK_FILENAME}.js`),
+            path.join(
+              SERVER_DIRECTORY,
+              `edge-${INSTRUMENTATION_HOOK_FILENAME}.js`
+            ),
+          ].filter((filePath) => existsSync(path.join(distDir, filePath)))
```

Comment:
nit: if we avoid `.filter` here we could avoid `existsSync`.

Reply:
Moved the manifest writing to later phase, and use the value that can determine if we already have edge routes. Then we don't need to do this fs check anymore!